[Lim Yuh Ching] iP

limyuhching opened the pull request.

limyuhching committed:
Level 0. Rename, Greet, Exit

limyuhching committed:
Level 1. Echo

limyuhching committed:
Renamed Duke. Added greetings.

limyuhching committed:
Renamed list as tasks. Added isDone flag and markAsDone, markAsUndone functions.

limyuhching committed:
Add ToDos, Events, Deadlines

ToDos, Events, Deadlines are created as a subclass of Task.

Moved addToTask and printTask from Task class to Main program.

TeoHaoZhi commented on src/main/java/Deadlines.java:
```diff
@@ -0,0 +1,18 @@
+public class Deadlines extends Task {
+    String by;
```
Good following of spacing requirements

TeoHaoZhi commented on src/main/java/Duke.java:
```diff
@@ -1,27 +1,89 @@
+    private static int taskCount = 0;
+
+    public static void addToTasks(String description, Type type) {
+        switch (type) {
```
Good and clear use of switch statements

TeoHaoZhi commented on src/main/java/Duke.java:
```diff
@@ -1,27 +1,89 @@
+        System.out.println("\tNow you have " + taskCount + " task(s) in the list.");
+    }
+    public static void printTasks() {
+        if (taskCount == 0) {
```
Good use of if-statements, no nesting errors

TeoHaoZhi reviewed (commented):
Overall code is easy to read, little to no standard coding violations. Everything looks good to me

TeoHaoZhi commented on src/main/java/Deadlines.java:
```diff
@@ -0,0 +1,18 @@
+        super(description);
+        this.by = by;
+    }
+    public String getBy() {
```
Good following of  naming of method conventions

TeoHaoZhi reviewed (commented):
Missed out 1 comment previously, everything looks good to me

DextheChik3n commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,95 @@
+            Scanner scanner = new Scanner(System.in);
+            String userInput = scanner.nextLine();
+            String[] substr = userInput.split("\\s+");
+            switch (substr[0]) {
```
command handling is well-organised using a switch case which makes it easier to troubleshoot.

DextheChik3n commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,95 @@
+            }
+        }
+    }
+    public static void bye() {
```
Perhaps you could use a more clear method name such as printByeMessage?
```suggestion
    public static void printByeMessage() {
```

DextheChik3n commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,95 @@
+            tasks[taskCount] = new ToDos(description);
+            break;
+        case DEADLINE:
+            int i = description.indexOf("/");
```
Maybe you can use a better variable name for the index like indexAfterBy? so that it is more clearer when you revisit the code

DextheChik3n reviewed (commented):
I really like how you structure your code! I just made some suggestions to the variable name to make your code more clear to understand

yicheng-toh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,95 @@
+            case "event":
+                addToTasks(userInput.substring(6), Type.EVENT);
+                break;
+            }
```
I like how you use switch statements for the various instructions. This makes your code very neat.

yicheng-toh commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,32 @@
+        this.description = description;
+    }
+
+    public boolean isDone() {
```
Perhaps the method could sound more verb-like?

```suggestion
    public boolean checkIsDone() {
```

yicheng-toh commented on src/main/java/Type.java:
```diff
@@ -0,0 +1,3 @@
+public enum Type {
+    TODO, DEADLINE, EVENT
+}
```
Good use of enum for the various tasks!

yicheng-toh reviewed (commented):
Looks good and neat!

limyuhching committed:
Make changes to various variable name

irving11119 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,95 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        // Greetings
+        String intro = "~~~~~~~~~~~~~~~~~~~\n"
```
Perhaps this could be made a constant.

irving11119 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,95 @@
+            tasks[taskCount] = new Deadlines(description, by);
+            break;
+        case EVENT:
+            int j = description.indexOf("/");
+            int k = description.indexOf("/", j+1);
```
For the following methods, you could use more descriptive variables rather than single letter variables. This would be more in line with the naming conventions required.

The name change suggested by your classmate would be a good example of how to name the variable instead!

irving11119 reviewed (commented):
Overall, your code follows the standards decently. However, do look out for things like naming conventions and magic types which can be avoided and will help to improve your code.

limyuhching committed:
Add Exceptions

limyuhching committed:
Revert "Add Exceptions"

This reverts commit 2e8d61b993b9164318c618422d2e99bef2e6b156.

limyuhching committed:
Handled exceptions

limyuhching committed:
Add support for deleting tasks from the list

limyuhching committed:
Add DataManager to save tasks to a text file

limyuhching committed:
Merge branch 'branch-Level-6' into branch-Level-7

# Conflicts:
#	src/main/java/Duke.java

limyuhching committed:
Merge branch 'branch-Level-6'

limyuhching committed:
Merge branch 'branch-Level-7'

limyuhching committed:
Divide code into smaller objects

Create TaskList, Parser, Ui and Storage to split the code in Duke into different objects

limyuhching committed:
Create find command to match keyword with task

limyuhching committed:
Add Javadoc to most of the class

limyuhching committed:
Merge pull request #1 from limyuhching/branch-A-JavaDoc

Add Javadoc to most of the class

limyuhching committed:
Merge branch 'master' of https://github.com/limyuhching/ip

limyuhching committed:
Update README.md

limyuhching committed:
Update README.md

limyuhching committed:
Merge branch 'master' of https://github.com/limyuhching/ip

limyuhching committed:
Minor UI changes

limyuhching committed:
Minor UI changes

limyuhching committed:
Handle exception when argument is missing

limyuhching committed:
Update README.md
